lightly/transforms/fda_transform.py

```diff
@@ -0,0 +1,246 @@
+from lightly.transforms.random_frequency_mask_transform import (
+    RandomFrequencyMaskTransform,
+)
+from lightly.transforms.rfft2d_transform import RFFT2DTransform
+from lightly.transforms.solarize import RandomSolarization
+from lightly.transforms.torchvision_v2_compatibility import torchvision_transforms as T
+
+
+class FDAView1Transform:
+    def __init__(
+        self,
+        # Random resized crop
+        input_size: int = 224,
+        min_scale: float = 0.08,
+        # Color jitter
+        cj_prob: float = 0.8,
+        cj_contrast: float = 0.4,
+        cj_bright: float = 0.4,
+        cj_sat: float = 0.2,
+        cj_hue: float = 0.1,
+        cj_strength: float = 1.0,
+        # Grayscale
+        random_gray_scale: float = 0.2,
+        # Gaussian blur
+        gaussian_blur: float = 1.0,
+        sigmas: Tuple[float, float] = (0.1, 2),
+        kernel_size: Optional[float] = 23,
+        # Amplitude rescale
+        ampl_rescale_range: Tuple[float, float] = (0.8, 1.75),
+        amp_rescale_prob: float = 0.2,
+        # Phase shift
+        phase_shift_range: Tuple[float, float] = (0.4, 0.7),
+        phase_shift_prob: float = 0.2,
+        # Random frequency mask
+        rand_freq_mask_range: Tuple[float, float] = (0.01, 0.1),
+        rand_freq_mask_prob: float = 0.5,
+        # Gaussian mixture mask
+        num_gaussians: int = 20,
+        std_range: Tuple[float, float] = (10, 15),
+        gm_prob: float = 0.2,
+        # Other
+        hf_prob: float = 0.5,
+        solarization_prob: float = 0.0,
+    ):
+        color_jitter = T.ColorJitter(
+            brightness=cj_strength * cj_bright,
+            contrast=cj_strength * cj_contrast,
+            saturation=cj_strength * cj_sat,
+            hue=cj_strength * cj_hue,
+        )
+
+        transform = [
+            T.RandomResizedCrop(size=input_size, scale=(min_scale, 1.0)),
+            T.ToTensor(),
+            RFFT2DTransform(),
+            T.RandomApply(
+                [AmplitudeRescaleTransform(range=ampl_rescale_range)],
+                p=amp_rescale_prob,
+            ),
+            T.RandomApply(
+                [PhaseShiftTransform(range=phase_shift_range)], p=phase_shift_prob
+            ),
+            T.RandomApply(
+                [RandomFrequencyMaskTransform(k=rand_freq_mask_range)],
+                p=rand_freq_mask_prob,
+            ),
+            T.RandomApply(
+                [GaussianMixtureMask(num_gaussians=num_gaussians, std_range=std_range)],
+                p=gm_prob,
+            ),
+            IRFFT2DTransform(shape=(input_size, input_size)),
+            T.ToPILImage(),
+            T.RandomApply([color_jitter], p=cj_prob),
+            T.RandomGrayscale(p=random_gray_scale),
+            GaussianBlur(kernel_size=kernel_size, sigmas=sigmas, prob=gaussian_blur),
+            RandomSolarization(prob=solarization_prob),
+            T.RandomHorizontalFlip(p=hf_prob),
```

Comment:
We usually apply these transforms in a different order, see: https://github.com/lightly-ai/lightly/blob/f33a90f5fd105c1c1d76abbe3976daa739157444/lightly/transforms/byol_transform.py#L45C1-L51C56

Reply:
I just need to reorganize them without adding additional transforms from byol? It will look like this?
```
T.RandomHorizontalFlip(p=hf_prob),
T.RandomApply([color_jitter], p=cj_prob),
T.RandomGrayscale(p=random_gray_scale),
GaussianBlur(kernel_size=kernel_size, sigmas=sigmas, prob=gaussian_blur),
RandomSolarization(prob=solarization_prob),
```

Reply:
Thanks for the update! Now only the random rotation transform is outstanding